src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -233,13 +238,18 @@ private Task RenderPageCoreAsync(IRazorPage page, ViewContext context)
             }
         }
 
-        private IRazorPage GetLayoutPage(ViewContext context, string layoutPath)
+        private IRazorPage GetLayoutPage(ViewContext context, string executingFilePath, string layoutPath)
         {
-            var layoutPageResult = _viewEngine.FindPage(context, layoutPath);
+            var layoutPageResult = _viewEngine.GetPage(executingFilePath, layoutPath, isPartial: true);
+            if (layoutPageResult.Page == null)
+            {
+                layoutPageResult = _viewEngine.FindPage(context, layoutPath, isPartial: true);
```

Comment:
All of the tests work fine. But I'm still working on the details of this `GetPage()` / `GetView()` then `FindPage()` / `FindView()` pattern. Errors may use an empty `SearchedLocations` collection if the `GetX()` method tries and the `FindX()` method doesn't.

Most likely fix is to check `!layoutPageResult.SearchedLocations.Any()` before entering this block. At most one of the `GetX()` and `FindX()` methods will set `SearchedLocations` to anything but `Enumerable.Empty<string>()`.